### DISCUSSION TITLE
GPIO Ports Raspberry Pi 3 or 4

### DISCUSSION THREAD

**Discussion by JEQT** [artisan-roaster-scope/artisan#1655] [Q&A]
I want to know if it is possible or if someone has managed to read the temperature data from the Raspberry GPIO ports. I would appreciate any help.

**Comment by hermetheuscoffee** [comment]
A Raspberry Pi doesn't have temperature sensors, so you would have to attach some type of sensor to the Pi to do something like this. While there are I2C sensors that would connect, I don't know if that's compatible with Artisan. But I know that a Yocto Thermocouple works with both Artisan and the RPi, so that's a pretty straightforward option to use.

**Comment by MAKOMO** [comment]
Support for Raspberry GPIO ports needs to be implemented. Volunteers?

**Comment by JEQT** [comment]
I think I found a way to monitor the temperature of a sensor through the GPIO ports to Artisan, doing this using the "Program" option in "Configuration->Device", I'm going to do some tests on this with a sensor, for now I just used flat data from a .py file as in the example, I'm going to try to interleave the GPIO ports with the .py program. Now, among other things, I wanted to know if anyone who has installed or used Artisan on Raspberry Pi3 Bookworm gets the following error and was able to find a solution, none of the menus in the top toolbar are displayed, they just turn color and nothing else. I would appreciate any help or advice on this matter.
